service_location/src/main/kotlin/org/thechance/service_location/endpoints/LocationRoutes.kt

```diff
@@ -0,0 +1,41 @@
+package org.thechance.service_location.endpoints
+
+
+import io.ktor.server.routing.*
+import io.ktor.server.websocket.*
+import io.ktor.websocket.*
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import org.koin.ktor.ext.inject
+import org.thechance.service_location.data.SocketHandler
+import org.thechance.service_location.data.model.LocationDto
+import org.thechance.service_location.data.model.WebSocketLocation
+import org.thechance.service_location.util.INVALID_LOCATION
+
+fun Route.locationRoutes() {
+
+    val socketHandler: SocketHandler by inject()
+
+    route("/location") {
+
+        webSocket("/sender/{tripId}") {
+            val tripId = call.parameters["tripId"]?.trim().orEmpty()
+            for (frame in incoming) {
```

Comment:
you can remove for each and if condition and replace it with 

while (true){
                    runCatching {
                        val location = receiveDeserialized<LocationDto>()

because your receive and send an object